pghoard/transfer.py

```diff
@@ -100,15 +102,159 @@ def operation(self):
 TransferQueue = Queue
 
 
+@dataclass
+class TransferIncrement:
+    total_bytes_uploaded: float
+    tracked_at: float = dataclasses.field(default_factory=time.monotonic)
+
+
+@dataclass
+class UploadEventProgress:
+    key: str
+    file_size: Optional[int]
+    file_type: FileType
+    increments: List[TransferIncrement] = dataclasses.field(default_factory=list)
+    started_at: float = dataclasses.field(default_factory=time.monotonic)
+
+
+class UploadEventProgressTracker(PGHoardThread):
+    CHECK_FREQUENCY = 5.  # check every 5 seconds for progress
+    WARNING_TIMEOUT = 5. * 60  # log a warning in case there is no progress during last 5 minutes
```

Comment:
:thinking:  I initially had it as an integer, but if I remember correctly mypy started complaining when I extended the class on tests `PatchedUploadEventProgressTracker` and set both constants to smaller factions because I didn't want tests to take that long... I tried patching UploadEventProgressTracker without extending the class but doesn't work due to our current test case structure.